pkg/cli: ensure consistent defaults regardless of TESTFLAGS=-v

knz opened the pull request:
Fixes #21355.
Fix suggested by @tamird.

Prior to this patch, CLI tests using the interactive code path
from `sql.go` would see different results depending on whether
the tests were run with `-v` or not, because the interactive
code path likes its defaults to differ based on whether the
output happens to go to a terminal.

This makes it hard to expect a fixed output in tests regardless
of the test parameters -- in fact, two tests were failing
because of this.

To alleviate this, this patch introduces a new helper function
`setCLIDefaultsForTests` to be used by tests that wish to
call into the interactive code path of `cockroach sql`.

Note: tests calling into `cockroach sql` with `RunWithArgs`
cannot utilize this function. To ensure stable test output,
these tests can use e.g. `cockroach sql -e` which currently
happens to determine the same parameter defaults regardless
of the terminal environment.

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/21365)
<!-- Reviewable:end -->

tamird commented on pkg/cli/sql_util_test.go:
```diff
@@ -98,12 +98,10 @@ func TestRunQuery(t *testing.T) {
+	setCLIDefaultsForTests()
 
 	// Non-query statement.
+	var b bytes.Buffer
```
This gets used for a few other statements, so I'd maybe leave it above the comment.

knz replied:
Aye, done.

tamird reviewed (approved):
Thanks @knz!

knz committed:
pkg/cli: ensure consistent defaults regardless of TESTFLAGS=-v

Fix suggested by @tamird.

Prior to this patch, CLI tests using the interactive code path
from `sql.go` would see different results depending on whether
the tests were run with `-v` or not, because the interactive
code path likes its defaults to differ based on whether the
output happens to go to a terminal.

This makes it hard to expect a fixed output in tests regardless
of the test parameters -- in fact, two tests were failing
because of this.

To alleviate this, this patch introduces a new helper function
`setCLIDefaultsForTests` to be used by tests that wish to
call into the interactive code path of `cockroach sql`.

Note: tests calling into `cockroach sql` with `RunWithArgs`
cannot utilize this function. To ensure stable test output,
these tests can use e.g. `cockroach sql -e` which currently
happens to determine the same parameter defaults regardless
of the terminal environment.

Release note: None

mberhault commented:
LGTM

knz commented:
Thanks to both!
